### PR TITLE
Split up developer's guide

### LINKED ISSUE #1860: Split up dev guide into multiple files
The developer's guide has become unweildy.

### PR DESCRIPTION
**Pull request by bryevdv**
issues: closes #1860

Had to scratch this itch.

### CODE REVIEW

**Comment by bryevdv** [comment]
@bokeh/dev these docs have been deployed to http://bokeh.pydata.org/en/test/ plan to merge this tomorrow morning CST.

**Comment by damianavila** [comment]
@bryevdv this is not conflicting because of the rearrangement, but you need to incorporate the changes in #1862 before merging this one... essentially add this:
```
* ``BOKEH_LOCAL_DOCS_CDN`` --- What version of BokehJS to use when you are working
    with sphinx docs locally.
```
LGTM to be merged after the change requested.

File: sphinx/source/docs/dev_guide/notes.rst [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+Environment Variables
+---------------------
+
+There are several environment variables that can be useful for developers:
+
+* ``BOKEH_BROWSER`` --- What browser to use when opening plots
+    Valid values are any of the browser names understood by the python
+    standard library webbrowser_ module.
+
```
Comment (bryevdv) [inline sphinx/source/docs/dev_guide/notes.rst]: @damianavila it's already here :)

File: sphinx/source/docs/dev_guide/notes.rst [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+* ``BOKEH_LOCAL_DOCS_CDN`` --- What version of BokehJS to use when building  sphinx docs locally.
```
Comment (bryevdv) [inline sphinx/source/docs/dev_guide/notes.rst]: @damianavila it's already here :)

Comment (damianavila) [inline sphinx/source/docs/dev_guide/notes.rst]: je... It seems I loaded the file yesterday and I forgot to reload it this morning. I swear I checked the file and I did not see it!!! Thanks!

**Pull request merged**